crypto/http/http_lib.c

```diff
@@ -107,7 +107,13 @@ int OSSL_parse_url(const char *url, char **pscheme, char **puser, char **phost,
         p = ++host_end;
     } else {
         /* look for start of optional port, path, query, or fragment */
-        host_end = strpbrk(host, ":/?#");
+        host_end = strchr(host, ':');
+        if (host_end == NULL)
+            host_end = strchr(host, '/');
+        if (host_end == NULL)
+            host_end = strchr(host, '?');
+        if (host_end == NULL)
+            host_end = strchr(host, '#');
```

Comment:
Why the change away from `strpbrk(3)`?

Another neater way to avoid `strpbrk(3)` would be:

```suggestion
        host_end += strcspn(host, ":/?#");
```